tests/libvmi/storage.go

```diff
@@ -79,6 +95,12 @@ func addVolume(vmi *v1.VirtualMachineInstance, volume v1.Volume) {
 	}
 }
 
+func addFilesystem(vmi *v1.VirtualMachineInstance, filesystem v1.Filesystem) {
+	if !filesystemExists(vmi, filesystem) {
```

Comment:
I see this here, and in many other places, we just silently ignore the problem, maybe it would be better to error or panic here if someone adds a Volume or Disk and some objects already exist?

But I suppose that is out of scope for this PR. But we should definitely discuss it.

Reply:
You are right, we should discuss this / change it in a separate PR.